src/compiler/path.ts

```diff
@@ -214,6 +214,12 @@ function getEncodedRootLength(path: string): number {
         return ~path.length; // URL: "file://server", "http://server"
     }
 
+    // Data: "data:application/typescript,42"
+    const match = path.match(/*regexp*/ /^(data:(?:[a-z0-9]+(?:\/[a-z0-9]+))?(?:;base64)?,)./);

```

Comment:
Typically we make these regexes global, and also use `.exec`. But, do we actually use regexes at all in this code?

Reply:
Aside: I like that that has the added benefit of not triggering the ESLint rule for `/*regexp*/`.

Reply:
That was the first thing I noticed; the lint rule probably needs an exception for that one.

But my question about using regexes at all still remains.

Reply:
Ah sorry I misinterpreted - what is the question? Going off the description in https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URLs I added a few test cases.

> ```plaintext
> data:[<mediatype>][;base64],<data>
> ```

Reply:
My question was roughly "is it unprecedented to be using regex for parsing?", given that most of this code is all string manipulation, plus a tiny bit of regex for prefix/substring matching, containing no calls to `.match` or `.exec`.

Reply:
Oh gotcha - in my personal uninformed opinion, no; data URIs can be pretty complex / edge-case-y. If you want TS to be accurate here then I would think a regex is needed. But if you're ok with just matching on a _mostly_ accurate `data:/` prefix then no I don't think it would be needed. Do you have a preference on accuracy vs. complexity here?